test/extractor/react.test.ts

```diff
@@ -0,0 +1,518 @@
+    const expected = [
+      { keyName: 'key1' },
+      { keyName: 'key2' },
+    ];
+
+    const code = `
+      React         .    createElement     (    T,     {     keyName: 'key1' })
+      React.createElement(T,{keyName:'key2'})
+    `;
+
+    const extracted = await extractKeys(code, FILE_NAME);
+    expect(extracted).toEqual(expected);
+  });
+
+  it('ignores dynamic values', async () => {
+    const expected = [
+      { keyName: 'key1' },
+    ];
+
+    const code = `
+      React.createElement(T, { keyName:'key1' }, someValue)
+    `;
+
+    const extracted = await extractKeys(code, FILE_NAME);
+    expect(extracted).toEqual(expected);
+  });
+});
+
+describe.each([ 'js', 'ts', 'jsx', 'tsx' ])('useTranslate (.%s)', (ext) => {
+  const FILE_NAME = `test.${ext}`
+
+  it('extracts from the t call', async () => {
+    const expected = [{ keyName: 'key1' }];
+
+    const code = `
+      function Test () {
+        const { t } = useTranslate()
+        t('key1')
+      }
+    `;
+
+    const extracted = await extractKeys(code, FILE_NAME);
+    expect(extracted).toEqual(expected);
+  });
+
+  it('extracts the default value from parameters', async () => {
+    const expected = [{ keyName: 'key1', defaultValue: 'default value' }];
+
+    const code = `
+      function Test () {
+        const { t } = useTranslate()
+        t('key1', { defaultValue: 'default value' })
+      }
+    `;
+
+    const extracted = await extractKeys(code, FILE_NAME);
+    expect(extracted).toEqual(expected);
+  });
+
+  it('extracts the namespace from parameters', async () => {
+    const expected = [{ keyName: 'key1', namespace: 'ns1' }];
+
+    const code = `
+      function Test () {
+        const { t } = useTranslate()
+        t('key1', { ns: 'ns1' })
+      }
+    `;
+
+    const extracted = await extractKeys(code, FILE_NAME);
+    expect(extracted).toEqual(expected);
+  });
+
+  it('keeps track of the namespace specified in useTranslate', async () => {
+    const expected = [{ keyName: 'key1', namespace: 'namespace' }];
+
+    const code = `
+      function Test () {
+        const { t } = useTranslate('namespace')
+        t('key1')
+      }
+    `;
+
+    const extracted = await extractKeys(code, FILE_NAME);
+    expect(extracted).toEqual(expected);
+  });
+
+  it('keeps track of the namespace specified in useTranslate (array)', async () => {
+    const expected = [{ keyName: 'key1', namespace: 'namespace1' }];
+
+    const code = `
+      function Test () {
+        const { t } = useTranslate([ 'namespace1', 'namespace2' ])
+        t('key1')
+      }
+    `;
+
+    const extracted = await extractKeys(code, FILE_NAME);
+    expect(extracted).toEqual(expected);
+  });
+
+  it('overrides the specified namespace if one is passed as parameter', async () => {
+    const expected = [{ keyName: 'key1', namespace: 'ns1' }];
+
+    const code = `
+      function Test () {
+        const { t } = useTranslate('namespace')
+        t('key1', { ns: 'ns1' })
```

Comment:
I think you are not covering the case with full object:
```ts
t({ key: 'key', ns: 'namespace', defaultValue: 'default' })
```

Reply:
Hm, I don't handle the `t('key', 'default', { ... })` case either... functions with multiple signatures............. 🙃 

I'm a bit bothered by the `key` attribute though, which differs from the `keyName` property the properties state machine is expecting 🤔

Reply:
Hmm, the t function needs to be multi signature, as it's the thing that you use all the time and it needs to be comfortable. `keyName` is used on react component as `key` can't be used.